lib/handler/connect.c

```diff
@@ -416,9 +416,6 @@ static void on_getaddr(h2o_hostinfo_getaddr_req_t *getaddr_req, const char *errs
         }
     }
 
-    /* Try connecting if possible. */
-    if (self->server_addresses.used == self->server_addresses.size)
-        return;
```

Comment:
Thank you for working on the fix. The problem statement sounds correct to me.

OTOH, I'm not sure if the proposed solution is correct.

Consider the following case:
1. ipv4 lookup succeeds
2. after `name_resolution_delay` milliseconds, we attempt to connect to that IPv4 address
3. while the connection attempt is ongoing, ipv6 lookup fails
In this case, we want the connection attempt to continue running. However, with the proposed solution it seems to me that we'd be closing the socket immediately.

I wonder if the correct fix is to move this if statement inside `if (self->sock != NULL) {` right below.

Reply:
I understand, thank you very much for the explanation!

I pushed a new version with the two lines moved inside the condition on there being a pending connection attempt, and I added comments that (to me) explain the three cases that the code handles.